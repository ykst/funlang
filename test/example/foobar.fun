fun iterate(obj)
    while obj.cond()
        obj.proc()
        obj.post()
    \__
\__

fun counter()
    var i = 0

    fun countUp()
        i = i + 1
        return i
    \_

    return countUp
\_

fun foobar(num)
    var i = 0
    var c = counter()

    fun cond()
        
        return i <= num
    \__

    fun proc()
        if i % 15 == 0
            $print("FooBar\n")
        elsif i % 3 == 0
            $print("Foo\n")
        elsif i % 5 == 0
            $print("Bar\n")
        else
            $print(i)
            $print("\n")
        \__
    \__

    fun post()
        i = c()
    \_
\__

$eval("iterate" ++ "(foobar{30})")
