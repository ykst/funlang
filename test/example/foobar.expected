FooBar
1
2
Foo
4
Bar
Foo
7
8
Foo
Bar
11
Foo
13
14
FooBar
16
17
Foo
19
Bar
Foo
22
23
Foo
Bar
26
Foo
28
29
FooBar
